modules/drtracker.lua

```diff
@@ -194,64 +248,84 @@ function DRTracker:DRFaded(unit, spellID)
-			end
-
-			lastFrame = frame
-
-			frame:Show()
-		else
-			frame:Hide()
-		end
-	end
-end
-
 function DRTracker:COMBAT_LOG_EVENT_UNFILTERED(event)
 	self:CombatLogEvent(event, CombatLogGetCurrentEventInfo())
 end
 
-function DRTracker:CombatLogEvent(event, timestamp, eventType, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, spellID, spellName, spellSchool, auraType)
+function DRTracker:CombatLogEvent(_, timestamp, eventType, hideCaster, sourceGUID, sourceName, sourceFlags, sourceRaidFlags, destGUID, destName, destFlags, destRaidFlags, spellID, spellName, spellSchool, auraType)
 	-- Enemy had a debuff refreshed before it faded
 	-- Buff or debuff faded from an enemy
-	if eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_REMOVED" then
-		if auraType == "DEBUFF" and DRData:GetSpellCategory(spellID) then
+	if eventType == "SPELL_AURA_APPLIED" or eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_REMOVED" then
+		local drCat, catTbl = DRData:GetSpellCategory(spellID)
+		if drCat and auraType == "DEBUFF" then
 			local unit = GladiusEx:GetUnitIdByGUID(destGUID)
 			if unit and self.frame[unit] then
-				self:DRFaded(unit, spellID)
+				local tbl = catTbl and catTbl or { drCat }
+				for _, drCat in pairs(tbl) do -- K: Use a for loop to catch auras that DR with multiple categories (primarily Ring of Frost/Deep Freeze)				
+					
+					-- K: Dynamic DR reset early if we have a full duration aura on _REFRESH / _APPLIED 
+					local tracked = (self.frame[unit] and self.frame[unit].tracker) and self.frame[unit].tracker[drCat] or nil
+					if tracked and (eventType == "SPELL_AURA_REFRESH" or eventType == "SPELL_AURA_APPLIED") then
+						if self:HasFullDurationAura(unit, sourceGUID, spellID) then
+							tracked.active = false
+						end
+					end
+					
+					self:DRFaded(unit, drCat, spellID, eventType)
+				end
+			end
+		end
+	end
+end
+
+function DRTracker:HasFullDurationAura(unit, sourceGUID, spellID)
+	local fullDuration = GladiusEx.Data.AuraDurations and GladiusEx.Data.AuraDurations[spellID] or nil
+	
+	if fullDuration then
+		local srcUnit = GladiusEx:GetUnitIdByGUID(sourceGUID)
+
+		for i=1, 40 do
+			local name, _, _, _, _, duration, _, unitCaster, _, _, secID, secSourceGUID = UnitAura(unit, i, "HARMFUL")
+			if not name then break end
+			if secID == spellID then
+				if ((not secSourceGUID or secSourceGUID == sourceGUID) or ((not unitCaster and not scrUnit) or (unitCaster and srcUnit and unitCaster == srcUnit))) then
+					-- K: Some classes/races have CC duration reduction effects, thus we have to check if the aura is at least longer than 50% of fullDuration
+					-- which would imply it's possibly reduced by effects - but at least not DRd (which would be less than or equal to 50%)
+					-- Note: No class/race/comp combo has the possibility to reduce a CC by 50% or more
+					if duration == fullDuration or duration > (fullDuration / 2) then

```

Comment:
The 2nd case encompasses the first. And this moght warrant a slight window

Reply:
The first one is (most likely) a CPU save since division is much more expensive than a single equality comparison. That's why I kept it.

Reply:
dividing something by two would (should) never involve a division, only a bitshift, but sure

Reply:
True, I'll remove it. A bitshift isn't that expensive.